### ISSUE TITLE
Test Error

### ISSUE THREAD

**Issue by louisscruz** [AngularClass/angular2-webpack-starter#249]
I'm getting an error when running `npm test`

`TypeError: 
at /Users/louisstephancruz/Sites/dd/spec-bundle.js:8768 <- webpack:///~/es7-reflect-metadata/dist/browser.js:130:0`

I'm not sure how to debug without more info. Any ideas?

The spec-bundle.js looks like this:

    Error.stackTraceLimit = Infinity;
    require('phantomjs-polyfill');
    require('es6-promise');
    require('es6-shim');
    require('es7-reflect-metadata/dist/browser');

    require('zone.js/lib/browser/zone-microtask.js');
    require('zone.js/lib/browser/long-stack-trace-zone.js');
    require('zone.js/dist/jasmine-patch.js');

    var testContext = require.context('./test', true, /\.spec\.ts/);
    var appContext = require.context('./src', true, /\.spec\.ts/);

    appContext.keys().forEach(appContext);
    testContext.keys().forEach(testContext);

    var domAdapter = require('angular2/src/platform/browser/browser_adapter');
    domAdapter.BrowserDomAdapter.makeCurrent();

**Comment by gdi2290** [comment]
there seems to be a problem with one of the decorators you're using

**Comment by gdi2290** [comment]
@louisscruz  I updated Reflect-Metadata to show errors

**Comment by louisscruz** [comment]
Ok. Now it's showing:

`  TypeError: targetDescriptor null is not an object
  at /Users/louisstephancruz/Sites/dd/spec-bundle.js:4715 <- webpack:///~/es7-reflect-metadata/dist/browser.js:130:0`

Any idea how to fix this?

**Comment by gdi2290** [comment]
is your code on github?

**Comment by louisscruz** [comment]
Here it is:

<https://github.com/louisscruz/dd>

**Comment by louisscruz** [comment]
Still scratching my head on this one. Any ideas?

**Comment by m1ch4ls** [comment]
@louisscruz  Commenting out defines in DefinePlugin fixed this issue for me.

before:
```
plugins: [
      new DefinePlugin({
        // Environment helpers
        'process.env': {
          'ENV': JSON.stringify(ENV),
          'NODE_ENV': JSON.stringify(ENV)
        },
        'global': 'window',
        // TypeScript helpers
        '__metadata': 'Reflect.metadata',
        '__decorate': 'Reflect.decorate'
      }),
      new ProvidePlugin({
        //'__metadata': 'ts-helper/metadata',
        //'__decorate': 'ts-helper/decorate',
        '__awaiter': 'ts-helper/awaiter',
        '__extends': 'ts-helper/extends',
        '__param': 'ts-helper/param',
        'Reflect': 'es7-reflect-metadata/dist/browser'
      })
```
after:
```
plugins: [
      new DefinePlugin({
        // Environment helpers
        'process.env': {
          'ENV': JSON.stringify(ENV),
          'NODE_ENV': JSON.stringify(ENV)
        },
        'global': 'window',
        // TypeScript helpers
        //'__metadata': 'Reflect.metadata',
        //'__decorate': 'Reflect.decorate'
      }),
      new ProvidePlugin({
        '__metadata': 'ts-helper/metadata',
        '__decorate': 'ts-helper/decorate',
        '__awaiter': 'ts-helper/awaiter',
        '__extends': 'ts-helper/extends',
        '__param': 'ts-helper/param',
        'Reflect': 'es7-reflect-metadata/dist/browser'
      })
    ]
```

**Comment by louisscruz** [comment]
I'm going to give this a shot tonight or tomorrow when I have access to my computer. Thanks!

Louis

**Comment by gdi2290** [comment]
yeah sorry it looks lke your repo was on an older version. I should keep track of the changes by changing the package version

**Comment by gdi2290** [comment]
I released version 1.0.0 so I'll start to follow versioned releases

**Comment by louisscruz** [comment]
That did it! Thank you!
